esp-hal/src/sdio.rs

```diff
@@ -0,0 +1,586 @@
+//! # Secure Digital I/O - Slave Mode
+//!
+//! ## Overiew
+//!
+//! The peripheral can be used to transfer data over the SDIO bus in `Slave`
+//! mode.
+
+use embedded_hal_sdmmc::{
+    CardMode,
+    CardType,
+    Common,
+    Device,
+    FifoStatus,
+    Reset,
+    SdBusWidth as BusWidth,
+    command::Command,
+    response::Response,
+    tuning::{TuningMode, TuningWidth},
+};
+
+use crate::{
+    gpio::{InputConfig, OutputConfig, Pull},
+    pac,
+};
+
+mod config;
+mod hinf;
+mod interrupt;
+mod pins;
+mod slc;
+mod slchost;
+mod state;
+mod timing;
+
+pub use config::Config;
+pub use hinf::{AnyHinf, HinfInfo, HinfInstance};
+pub use interrupt::{DeviceInterrupt, HostInterrupt};
+pub use pins::Pins;
+pub use slc::{AnySlc, SlcInfo, SlcInstance};
+pub use slchost::{AnySlchost, SlchostInfo, SlchostInstance};
+pub use state::State;
+pub use timing::Timing;
+
+/// SDIO peripheral instance.
+pub trait PeripheralInstance: crate::private::Sealed {
```

Comment:
I'd prefer not having this trait, it's just extra indirection in the code that doesn't really do anything for us, it seems.

Reply:
I2C is probably the best one right now, but none of our drivers need to use multiple IPs at the same time. I think you might have looked at SPI which currently has an absolutely redundant PeripheralInstance trait that is being cleaned up in one of my draft PRs :)